pwiz_tools/Skyline/Model/SrmDocument.cs

```diff
@@ -2115,12 +2115,26 @@ public double GetCollisionEnergy(PeptideDocNode nodePep, TransitionGroupDocNode
         public static double GetCollisionEnergy(SrmSettings settings, PeptideDocNode nodePep,
             TransitionGroupDocNode nodeGroup, TransitionDocNode nodeTran, CollisionEnergyRegression regression, int step)
         {
-            var ce = nodeTran==null // If we're only given a precursor, use the explicit CE of its children if they all agree
-                ? (nodeGroup.Children.Any() && nodeGroup.Children.All( node => ((TransitionDocNode)node).ExplicitValues.CollisionEnergy == ((TransitionDocNode)nodeGroup.Children.First()).ExplicitValues.CollisionEnergy) 
-                    ? ((TransitionDocNode)nodeGroup.Children.First()).ExplicitValues.CollisionEnergy : null)
-                : nodeTran.ExplicitValues.CollisionEnergy;
+            double? ce = null;
+            // Collision Energy explicitly declared at the transition level is taken to be the correct value.
+            if (nodeTran != null)
+            {
+                ce = nodeTran.ExplicitValues.CollisionEnergy;
+            }
+            // If we're only given a precursor, use the explicit CE of its children if they all agree.
+            if (!ce.HasValue)
+            {
+                if (nodeGroup.Children.Select(node => ((TransitionDocNode)node).ExplicitValues.CollisionEnergy).Distinct().Count() == 1)
+                    ce = ((TransitionDocNode)nodeGroup.Children.First()).ExplicitValues.CollisionEnergy;
+            }
+            // Failing that, explicitly declared value at the precursor level is used.
+            if (!ce.HasValue)
+            {
+                ce = nodeGroup.ExplicitValues.CollisionEnergy;
+            }

```

Comment:
I would still prefer to see it as a function which would make this a single line double? ce = GetCE()

Reply:
Done.